tests/functional/platform_aliases/00-simple.t

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env bash
+# THIS FILE IS PART OF THE CYLC SUITE ENGINE.
+# Copyright (C) NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#-------------------------------------------------------------------------------
+# Test that a different platform is selected on different tries with the same
+# platform Alias.
+#
+# Because this test can only ever test that all platforms within an alias have
+# been used it is theoretically possible for the random selection from Python's
+# random.choice to continue selecting the same platform forever, which is the
+# same outcome as failure. As a result this test is deliberately slightly flaky:
```

Comment:
Should this not go under `tests/flakyfunctional/` then?

Reply:
Perhaps - I'm hoping that < 1% chance of flaking is probably ok... but happy to move it.